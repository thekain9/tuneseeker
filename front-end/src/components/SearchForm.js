import React, { useState } from 'react';  // Import React and useState from 'react' library
import { useNavigate } from 'react-router-dom';  // Import useNavigate from 'react-router-dom'
import '../styles/SearchForm.css';  // Import CSS styles

const SearchForm = () => {  // Define the SearchForm functional component
  const [term, setTerm] = useState('');  // Declare state variable 'term' and its setter function
  const [media, setMedia] = useState('all');  // Declare state variable 'media' and its setter function, the default will be "all"
  const navigate = useNavigate();  // Initialise the navigate function using useNavigate hook

  const handleSearch = () => {  // Define the handleSearch function
    navigate(`/api/search/${term}/gb/${media}`);  // Use navigate to redirect to a search URL
  };

  const handleKeyDown = (event) => {
    if (event.key === 'Enter') {
      handleSearch();
    }
  };
  
  return (
    <div className="search-page">
      <div className="search-bar">
        <h1 className='search-word'>Search</h1>
        <div className="search-input">
          <input
            type="text"
            placeholder="Search content"
            value={term}
            onChange={(e) => setTerm(e.target.value)}
            onKeyDown={handleKeyDown}
          />
          <select
            value={media}
            onChange={(e) => setMedia(e.target.value)}
          >
            <option value="movie">Movie</option>
            <option value="podcast">Podcast</option>
            <option value="music">Music</option>
            <option value="musicVideo">Music Video</option>
            <option value="audiobook">Audiobook</option>
            <option value="shortFilm">Short Film</option>
            <option value="tvShow">TV Show</option>
            <option value="software">Software</option>
            <option value="ebook">Ebook</option>
            <option value="all">All</option>
          </select>
          <button onClick={handleSearch} className='search-button'>Search</button>
        </div>
      </div>
      <h1 className="title">TUNESEEKER</h1>
    </div>
  );
};

export default SearchForm;


