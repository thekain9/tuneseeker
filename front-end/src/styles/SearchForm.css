body {

  background-color: #7897BF;
  
}

.search-page {
  
    font-family: 'Oswald', sans-serif;
    font-weight: 700;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    min-height: 100vh;
    text-align: center;
    overflow: hidden;
    margin: 0;
    padding: 0;
  }
  
  .search-bar {

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex-grow: 1;
    text-align: center;
    margin-bottom: 0px;
  }
  
  .search-input {
    font-family: 'Oswald', sans-serif;
    font-weight: 400;
    display: flex;
    flex-direction: row;
    align-items: center;

  }
  h1 {
    font-family: 'Oswald', sans-serif;
    font-weight: 700;
    font-size: 5em;
  }

  input, select, option, button {
    font-family: 'Oswald', sans-serif;
    font-family: 400;
    font-size: 1.2em;
    color: #F2F0E9;
    background: #D96E48;
    border: 0;
    border-radius: 5px;
    margin: .8em;
  }

  .search-page .search-bar input::placeholder {
    font-family: 'Oswald', sans-serif;
    font-weight: 400;
    color: #F2EFBB; 
  }
  
  .title {
    font-family: 'Oswald', sans-serif;
    font-weight: 700;
    font-size: 20.5em;
    margin: auto 0 0; 
    padding: 0;
    margin-bottom: 0;
  }

  