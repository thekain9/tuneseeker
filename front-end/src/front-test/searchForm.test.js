// Importing necessary React library.
import React from 'react';

// Importing functions from the testing library: 'render' to render React components,
// and 'fireEvent' to simulate various DOM events on elements.
import { render, fireEvent } from '@testing-library/react';

// Importing the BrowserRouter from 'react-router-dom' and renaming it as 'Router'.
// This provides routing capabilities for React applications.
import { BrowserRouter as Router } from 'react-router-dom';

// Importing the SearchForm component from its location to be tested.
import SearchForm from '../components/SearchForm';

// Snapshot test:
// This test will check if the rendered output of the SearchForm component matches
// a previously stored snapshot. If there's a difference, the test will fail.
// It helps ensure that unintentional UI changes aren't introduced.
test('SearchForm snapshot', () => {
  // Rendering the SearchForm component within a Router because it uses 
  // routing-specific hooks. The 'asFragment' function captures the output.
  const { asFragment } = render(
    <Router>
      <SearchForm />
    </Router>
  );
  
  // This assertion checks if the current rendered output matches a saved snapshot.
  expect(asFragment()).toMatchSnapshot();
});

// Unit test:
// This test simulates user interaction by changing the value of an input field 
// and then checks if the new value was correctly set.
test('changes input value', () => {
  // Render the SearchForm component to access its elements.
  const { getByPlaceholderText } = render(
    <Router>
      <SearchForm />
    </Router>
  );

  // Accessing the input element using its placeholder text.
  const input = getByPlaceholderText('Search content');

  // Simulating a change event on the input element, setting its value to 'test'.
  fireEvent.change(input, { target: { value: 'test' } });
  
  // Assertion to check if the input's value was correctly set to 'test'.
  expect(input.value).toBe('test');
});

