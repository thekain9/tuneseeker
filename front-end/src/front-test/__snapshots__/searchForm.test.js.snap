// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-page"
  >
    <div
      class="search-bar"
    >
      <h1
        class="search-word"
      >
        Search
      </h1>
      <div
        class="search-input"
      >
        <input
          placeholder="Search content"
          type="text"
          value=""
        />
        <select>
          <option
            value="movie"
          >
            Movie
          </option>
          <option
            value="podcast"
          >
            Podcast
          </option>
          <option
            value="music"
          >
            Music
          </option>
          <option
            value="musicVideo"
          >
            Music Video
          </option>
          <option
            value="audiobook"
          >
            Audiobook
          </option>
          <option
            value="shortFilm"
          >
            Short Film
          </option>
          <option
            value="tvShow"
          >
            TV Show
          </option>
          <option
            value="software"
          >
            Software
          </option>
          <option
            value="ebook"
          >
            Ebook
          </option>
          <option
            value="all"
          >
            All
          </option>
        </select>
        <button
          class="search-button"
        >
          Search
        </button>
      </div>
    </div>
    <h1
      class="title"
    >
      TUNESEEKER
    </h1>
  </div>
</DocumentFragment>
`;
